src/main/java/com/algotrading/backtesting/stock/Stock.java

```diff
@@ -73,6 +77,10 @@ public boolean readFromMongoDB() {
 		for (Ticker ticker : tickers) {
 			try {
 				Date date = sdf.parse(ticker.date);
+				if( earliestDate.equals( null ) || earliestDate.compareTo( date ) > 0 )

```

Comment:
I wonder is it better to sort the tickers and get the first and last will be cleaner, the number of element is not that much, sorting should not be slow